css/css-align/self-alignment/block-justify-self.html

```diff
@@ -0,0 +1,199 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Self-Alignment: justify-self - block-level elements</title>
+<link rel="author" title="David Tran" href="mailto:davidtranhq+wpt@gmail.com">
+<link rel="match" href="block-justify-self-ref.html">
+<link rel="help" href="https://www.w3.org/TR/css-align-3/#justify-block" />
+<meta name="assert" content="justify-self applies to block-level boxes">
+<style>
+.grid-container div {
+  width: 100px;
+  background-color: green;
+}
+
+.block-container div, .block-container-with-table-items div {
+  width: 100px;
+  background-color: red;
+}
+
+.stack {
+  display: grid;
+}
+
+.stack > * {
+  grid-row: 1;
+  grid-column: 1;
+}
+
+body {
+  width: 700px;
+}
+
+</style>
+<p>Test passes if <strong>no red</strong> is visible</p>
+
+<div class="stack" style="float: left;">
+  <article class="block-container" style="display: block; width: 300px">
+    <div style="justify-self: auto;">auto</div>
+    <div style="justify-self: normal;">normal</div>
+    <div style="justify-self: stretch; width: auto; margin: 0;">stretch</div>
+    <div style="justify-self: stretch; margin: 0;">stretch</div>
+    <div style="justify-self: stretch; width: auto; margin: auto;">stretch</div>
+    <div style="justify-self: left;">left</div>
+    <div style="justify-self: right;">right</div>
+    <div style="justify-self: start;">start</div>
+    <div style="justify-self: center;">center</div>
+    <div style="justify-self: end;">end</div>
+    <div style="justify-self: self-start;">self-start</div>
+    <div style="justify-self: self-end;">self-end</div>
+    <div style="justify-self: flex-start;">flex-start</div>
+    <div style="justify-self: flex-end;">flex-end</div>
+    <div style="float: left; justify-self: center;">float left</div>
```

Comment:
But the block container doesn't ignore the `float` property. In fact, https://drafts.csswg.org/css-align/#justify-block says about `justify-self`: "This property does not apply to floats."

So when a grid child has both `float` and `justify-self`, `justify-self` wins.

But when a block flow child has both `float` and `justify-self`, `float` wins.

At least this is my interpretation. So I think your -ref file needs something different for these `float`s.

But @fantasai, please let me know if I've misunderstood.

Reply:
@davidtranhq I see that in your commit message you have "functionality is disabled on display: float" which indicates we agree on the intent of the spec. So I suppose you just have to fix the -ref grid item(s) that correspond to these floats.

Sorry I missed that the first time around.

Reply:
Sorry for the late reply, I was working internally on another version of these tests and missed this. I believe I have corrected the issues you were talking about and will make another PR for it :)